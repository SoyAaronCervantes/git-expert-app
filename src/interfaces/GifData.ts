export interface GifsData {
  id: string;
  title: string;
  url: string;
}
