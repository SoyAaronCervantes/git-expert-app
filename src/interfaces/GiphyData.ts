export interface GiphyData {
  analytics: GiphyDataAnalytics;
  analytics_response_payload: string;
  bitly_gif_url: string;
  bitly_url: string;
  content_url: string;
  embed_url: string;
  id: string;
  images: GiphyImages;
  import_datetime: string;
  is_sticker: number;
  rating: string;
  slug: string;
  source: string;
  source_post_url: string;
  source_tld: string;
  title: string;
  trending_datetime: string;
  type: string;
  url: string;
}


interface GiphyDataAnalytics {
  onclick: GiphyDataAnalyticsEvents;
  onload: GiphyDataAnalyticsEvents;
  onsent: GiphyDataAnalyticsEvents;
}

interface GiphyDataAnalyticsEvents {
  url: string;
}

interface GiphyImages {
  "480w_still": GiphyImagesData;
  downsized: GiphyImagesData;
  downsized_large: GiphyImagesData;
  downsized_medium: GiphyImagesData;
  downsized_small: GiphyImagesData;
  downsized_still: GiphyImagesData;
  fixed_height: GiphyImagesData;
  fixed_height_downsampled: GiphyImagesData;
  fixed_height_small: GiphyImagesData;
  fixed_height_small_still: GiphyImagesData;
  fixed_height_still: GiphyImagesData;
  fixed_width: GiphyImagesData;
  fixed_width_downsampled: GiphyImagesData;
  fixed_width_small: GiphyImagesData;
  fixed_width_small_still: GiphyImagesData;
  fixed_width_still: GiphyImagesData;
  looping: GiphyImagesData;
  original: GiphyImagesData;
  original_mp4: GiphyImagesData;
  original_still: GiphyImagesData;
  preview: GiphyImagesData;
  preview_gif: GiphyImagesData;
  preview_webp: GiphyImagesData;
};

interface GiphyImagesData {
  width: string;
  size: string;
  url: string;
  height: string;
}
