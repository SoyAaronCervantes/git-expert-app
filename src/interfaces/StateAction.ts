import {Dispatch, SetStateAction} from "react";

export interface StateAction<T> {
  [ value: string ]: Dispatch<SetStateAction<T>>
}