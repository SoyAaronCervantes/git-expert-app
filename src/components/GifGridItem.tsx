import React from 'react';
import PropTypes from 'prop-types';
import {GifsData} from "../interfaces/GifData";

const GifGridItem = ({ item }: { item: GifsData }) => {
  return (
    <div className={ "card" }>
      <img src={ item.url } alt={ item.title }  />
      <p>{ item.title }</p>
    </div>
  );
};

GifGridItem.propTypes = {
  item: PropTypes.object.isRequired
};

export default GifGridItem;
