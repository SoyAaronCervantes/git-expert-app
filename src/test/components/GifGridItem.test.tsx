import React, {ImgHTMLAttributes} from 'react';
import { shallow } from 'enzyme';
import GifGridItem from '../../components/GifGridItem';
import {GifsData} from '../../interfaces/GifData';

describe('Test on <GifGridItem />', () => {
  const data: GifsData = { id: 'strongID', title: 'oneplus', url: 'https://url.com' }

  let wrapper = shallow( <GifGridItem item={ data } /> )
  beforeEach( () => wrapper = shallow( <GifGridItem item={ data } /> ) )

  test('display <GifFridItem /> correctly', () => {
    expect( wrapper ).toMatchSnapshot();
  });

  test('should have a paragraph using title property', () => {
    const p = wrapper.find('p');
    expect( p.text().trim() ).toBe( data.title );
  });

  test( 'should use title property on alt property and get an image from url property', () => {
    const img = wrapper.find('img');
    const imgProps: ImgHTMLAttributes<any> = img.props();
    expect( imgProps.src ).toBe(data.url );
    expect( imgProps.alt ).toBe(data.title );
   })

  test( 'should have "animate__fadeIn" on container', () => {
    const div = wrapper.find('div');
    const divProps = div.props();
    expect( divProps.className ).toMatch('animate__fadeIn')
  })

});