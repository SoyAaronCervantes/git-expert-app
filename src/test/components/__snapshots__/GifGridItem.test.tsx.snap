// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifGridItem /> display <GifFridItem /> correctly 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="oneplus"
    src="https://url.com"
  />
  <p>
    oneplus
  </p>
</div>
`;
